storage/Devices/BlkDeviceImpl.h

```diff
@@ -211,6 +221,13 @@ namespace storage
      */
     void wait_for_devices(const vector<const BlkDevice*>& blk_devices);
 
+
+    /**
+     * Run "udevadm settle" and check non existence of all blk devices.
+     */
+    void wait_for_detach_devices(const vector<const BlkDevice*>& blk_devices);
```

Comment:
Should this really be an official part of the API? This sounds very much like an implementation detail to me.

Reply:
Well, this is declared in a Impl header, so I expect this does not belong to the public API. Anyway, basically I have mimic the method `wait_for_devices` that is already there. This new `wait_for_detach_devices` is just the opposite method, it waits to devices disappear instead of waiting for devices being created.

Reply:
Ah, duh, silly me; I thought I had checked that this is the public header. Disregard.